test/TestBase.py

```diff
@@ -190,4 +203,28 @@ def assertAccessDependency(self,
             self.remove(items)
 
     def assertBeatable(self, beatable: bool):
+        """Asserts that the game can be beaten with the current state"""
         self.assertEqual(self.multiworld.can_beat_game(self.multiworld.state), beatable)
+
+    # following tests are automatically run
+    def testAllStateCanReachEverything(self):
+        """Ensure all state can reach everything with the defined options"""
+        if not (hasattr(self, "game") and hasattr(self, "multiworld") and getattr(self, "options", {})):
+            return
+        with self.subTest("Game", game=self.game):
+            excluded = self.multiworld.exclude_locations[1].value
+            state = self.multiworld.get_all_state(False)
+            for location in self.multiworld.get_locations():
+                if location.name not in excluded:
+                    with self.subTest("Location should be reached", location=location):
+                        self.assertTrue(location.can_reach(state), f"{location.name} unreachable")
+
+    def testEmptyStateCanReachSomething(self):
+        """Ensure empty state can reach at least one location with the defined options"""
+        if not (hasattr(self, "game") and hasattr(self, "multiworld") and getattr(self, "options", {})):
```

Comment:
what is the `and getattr(self, "options", {}))` exactly for?
i am assuming it is to skip the test if everything is default, because that test is already run from the core test - this is only partially correct, because one could override `world_setup` or `setUp`.

My first idea was to check if `auto_construct` is False to detect a customized setup, but might also be incomplete since you could use regular `auto_construct` with a custom `setup_world`.

Reply:
```python
        constructed = hasattr(self, "game") and hasattr(self, "multiworld")
        if not constructed or (not self.options
                               and self.setUp is WorldTestBase.setUp
                               and self.world_setup is WorldTestBase.world_setup):
            return  # not possible or identical to the base test that's always being run already
```
maybe?